### PR TITLE
Synthesis pre processing

### PR DESCRIPTION
**Pull request by ShiriMoran**

### CODE REVIEW

File: pkg/collector/data_model.go [hunk 1]
```diff
@@ -431,6 +431,10 @@ func (group *Group) UnmarshalJSON(b []byte) error {
 	)
 }
 
+func (group *Group) Name() string {
+	return *group.Group.DisplayName
```
Comment (adisos) [inline pkg/collector/data_model.go:435]:
```suggestion
	if group.Group.DisplayName == nil{
	  return ""
	}
	return *group.Group.DisplayName
```

File: pkg/synthesis/synthesis.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package synthesis
+
+import (
+	"fmt"
+
+	"github.com/np-guard/vmware-analyzer/pkg/collector"
+	"github.com/np-guard/vmware-analyzer/pkg/model"
+)
+
+func NSXSynthesis(recourses *collector.ResourcesContainerModel, params model.OutputParameters) (string, error) {
```
Comment (adisos) [inline pkg/synthesis/synthesis.go:10]:
```suggestion
func NSXToAbstractModelSynthesis(recourses *collector.ResourcesContainerModel, params model.OutputParameters) (string, error) {
```

File: pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+category: Application
```
Comment (adisos) [inline pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt:1]: is there option to override expected output with new output?

File: pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+symbolic inbound rules:
+	0. action: allow paths: (group  = DumbledoreAll) to (*)
```
Comment (adisos) [inline pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt:4]: is rule scope / category scope going to be reflected in symbolic version?
or is it going to be combined as src/dst symbolic expression?

File: pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt (hunk 3 above)
Comment (ShiriMoran) [reply pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt:1]: We can change the file. Or is there something else you had in mind?

File: pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt (hunk 4 above)
Comment (ShiriMoran) [reply pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt:4]: The latter. 
Pointer to the original rule is kept, but that's it.

Regarding the scope - basically its just another term that can be differed to the last stages before the actual synthesis.

File: pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt (hunk 3 above)
Comment (adisos) [reply pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt:1]: code that updates those files automatically when needed?

Comment (ShiriMoran) [reply pkg/synthesis/tests_expected_output/ExampleDumbeldore.txt:1]: Opened an issue 
#104

**Review by adisos — APPROVED** [review]

**Pull request merged**